validator/sawtooth_validator/server/events/subscription.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from abc import ABCMeta
+from abc import abstractmethod
+import re
+
+
+class EventSubscription:
+    """Represents a subscription to events. An event is part of a subscription
+    if its type matches the type of the subscription and, if any filters are
+    included in the subscription, it passes all filters.
+    """
+    def __init__(self, event_type, filters=None):
+        self.event_type = event_type
+        if filters:
+            self.filters = filters
+        else:
+            self.filters = []
+
+    def __eq__(self, other):
+        if self.event_type != other.event_type:
+            return False
+
+        for f in self.filters:
+            if f not in other.filters:
+                return False
+
+        return True
+
+    def __contains__(self, event):
+        """Returns whether this events belongs within this subscriptions."""
+        if event.event_type == self.event_type:
+            for sub_filter in self.filters:
+                if event not in sub_filter:
+                    return False
+            return True
+
+        return False
+
+
+class InvalidFilterError(Exception):
+    pass
+
+
+class EventFilterType:
```

Comment:
You could define FilterType as a high-level protobuf enum (similar to RoleType in authroization.proto) to remove the need for this class.

Reply:
I wanted to keep the representation of filters within the validator decoupled from the protobuf definitions used when subscribing to them, so I didn't want to import anything from that file here.